synapse/http/matrixfederationclient.py

```diff
@@ -473,8 +473,6 @@ async def _send_request(
                             )
 
                             response = await request_deferred
-                    except TimeoutError as e:
-                        raise RequestSendFailed(e, can_retry=True) from e
```

Comment:
this was never actually being hit, because it was catching the wrong type of TimeoutError, but in any case it's redundant since the `except Exception` clause below is the same.

(I'm not really sure why the additional clause was added. I think it was to avoid some logging, which was later removed in https://github.com/matrix-org/synapse/commit/a0f574f3c291374c11b7b1cac8fc7c4fbb6ef7b2#diff-eaf1ef90b23076afc24efff2afdfe772L439.)